Framework/CurveFitting/src/Functions/AsymmetricPearsonVII.cpp

```diff
@@ -0,0 +1,214 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+//----------------------------------------------------------------------
+// Includes
+//----------------------------------------------------------------------
+#include "MantidCurveFitting/Functions/AsymmetricPearsonVII.h"
+#include "MantidAPI/FunctionFactory.h"
+#include "MantidCurveFitting/Constraints/BoundaryConstraint.h"
+
+#include <cmath>
+#include <iostream>
+
+namespace Mantid::CurveFitting::Functions {
+
+using namespace CurveFitting;
+using namespace Constraints;
+using namespace std;
+using namespace Kernel;
+using namespace API;
+
+DECLARE_FUNCTION(AsymmetricPearsonVII)
+
+AsymmetricPearsonVII::AsymmetricPearsonVII() : IPeakFunction() {}
+
+void AsymmetricPearsonVII::init() {
+  declareParameter("PeakHeight", 0.0, "Hight of the peak");
+  declareParameter("PeakCentre", 0.0, "Location of the peak");
+  declareParameter("Width", 0.0, "Full width at half maximum");
+  declareParameter("LeftShape", 1.0, "Left shape parameter");
+  declareParameter("RightShape", 1.0, "Right shape parameter");
+
+  // constrain shape parameters to the suggested range of values
+  auto leftShapeConstraint = std::make_unique<BoundaryConstraint>(this, "LeftShape", 1.0, 20.0, true);
+  addConstraint(std::move(leftShapeConstraint));
+  auto rightShapeConstraint = std::make_unique<BoundaryConstraint>(this, "RightShape", 1.0, 20.0, true);
+  addConstraint(std::move(rightShapeConstraint));
+}
+
+void AsymmetricPearsonVII::functionLocal(double *out, const double *xValues, const size_t nData) const {
+  const double peak_height = getParameter("PeakHeight");
+  const double peak_centre = getParameter("PeakCentre");
+  const double weight = 1.0 / getParameter("Width");
+  const double ml = getParameter("LeftShape");
+  const double mr = getParameter("RightShape");
+  double weight_sq = weight * weight;
+
+  for (size_t i = 0; i < nData; ++i) {
+    double offset = xValues[i] - peak_centre;
+    double offset_sq = offset * offset;
+    double left_part, right_part;
+
+    if (ml == 0.0)
```

Comment:
does line 37 make this unable to go below 1?

Reply:
As far as I understand, that line sets the default minimum value for ml to be 1 (as requested by users). However, a user can manually replace that value to be anything in the [0, infty) range, so in principle ml can be set to 0.